src/argilla/monitoring/asgi.py

```diff
@@ -183,21 +207,30 @@ async def _extract_response_content(
             body = response.body
         return new_response, json.loads(body)
 
-    def _log_to_argilla(
-        self,
-        inputs: List[Dict[str, Any]],
-        outputs: List[Dict[str, Any]],
-        url: str,
-        **tags
+    def _prepare_argilla_data(
+        self, inputs: List[Dict[str, Any]], outputs: List[Dict[str, Any]], **tags
     ):
-        records = [
-            record
-            for _inputs, _outputs in zip(inputs, outputs)
-            for record in [self._records_mapper(_inputs, _outputs)]
-            if record
-        ]
-
-        if records:
-            for r in records:
-                r.prediction_agent = url
-            argilla.log(records=records, name=self._dataset, tags=tags)
+        # using the base monitor, we only need to provide the input data to the rg.log function
+        # and the monitor will handle the sample rate, queue and argilla interaction
+        try:
+            records = self._records_mapper(inputs, outputs)
+            assert records, ValueError(
+                "The records_mapper returns and empty record list."
+            )
+            if not isinstance(records, list):
+                records = [records]
+        except Exception as ex:
+            records = []
+            _logger.error(
+                "Cannot log to argilla. Error in records mapper.", exc_info=ex
+            )
+
+        for record in records:
+            if self._monitor.agent is not None:
+                record.prediction_agent = self._monitor.agent
```

Comment:
In case that record mapper already set the agent:
```suggestion
            if self._monitor.agent is not None and not record.prediction_agent:
                record.prediction_agent = self._monitor.agent
```